tests/create_time_intervals_data.py

```diff
@@ -0,0 +1,44 @@
+import shutil
+from contextlib import contextmanager
+from pathlib import Path
+
+import numpy as np
+import yaml
+
+TIME_INTERVAL_ROOT = Path("tests/time_interval_data")
+
+
+@contextmanager
+def create_time_intervals_data():
```

Comment:
this is a nice start but I think we need a couple of more test cases where
1. the intervals are irregular (right now everything is in steps of 200)
2. the intervals are overlapping 
    * 2 overlapping together 
    * 3 overlapping together
    * a break time in between (eg not train-test-val between 450 and 500)
    
@oliveira-caio could we please add these?